jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -767,6 +767,222 @@ ecma_builtin_typedarray_prototype_set (ecma_value_t this_arg, /**< this argument
+  }
+
+  if (ecma_is_value_undefined (index_value)
+     || ecma_is_value_null (index_value))
+  {
+    ret_value = ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
+  }
+  else
+  {
+    ret_value = ecma_op_to_string (index_value);
+  }
+
+  ecma_free_value (index_value);
+  ecma_deref_ecma_string (index_string_p);
+  return ret_value;
+} /* ecma_op_typedarray_get_to_string_at_index */
+
+/**
+ * The TypedArray.prototype.toString's separator creation routine based on
+ * the Array.prototype.toString's separator routine
+ *
+ * See also:
+ *          ECMA-262 v5.1, 15.4.4.2 4th step
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_op_typedarray_get_separator_string (ecma_value_t separator) /**< possible separator */
+{
+  if (ecma_is_value_undefined (separator))
+  {
+    return ecma_make_magic_string_value (LIT_MAGIC_STRING_COMMA_CHAR);
+  }
+
+  return ecma_op_to_string (separator);
+} /* ecma_op_typedarray_get_separator_string */
+
+/**
+ * The TypedArray.prototype object's 'join' routine basen on
+ * the Array.porottype object's 'join'
+ *
+ * See also:
+ *          ECMA-262 v5, 15.4.4.5
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_join (const ecma_value_t this_arg, /**< this argument */
+                                        const ecma_value_t separator_arg) /**< separator argument */
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  /* 1. */
+  ecma_value_t obj_value = ecma_op_to_object(this_arg);
+
+  if (ECMA_IS_VALUE_ERROR (obj_value))
+  {
+    return obj_value;
+  }
+  ecma_object_t *obj_p = ecma_get_object_from_value (obj_value);
+
+  /* 2. */
+  ecma_value_t length_value = ecma_op_object_get_by_magic_id(obj_p, LIT_MAGIC_STRING_LENGTH);
+
+  if (ECMA_IS_VALUE_ERROR (length_value))
+  {
+    return length_value;
+  }
+
+  ECMA_OP_TO_NUMBER_TRY_CATCH (length_number,
+                              length_value,
+                              ret_value);
+
+  /* 3. */
+  uint32_t length = ecma_number_to_uint32 (length_number);
+
+  /* 4-5. */
+  ecma_value_t separator_value = ecma_op_typedarray_get_separator_string (separator_arg);
+  if( ECMA_IS_VALUE_ERROR (separator_value))
+  {
+    return separator_value;
+  }
+  if (length == 0)
+
+  {
+    /* 6. */
+    ret_value = ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
+  }
+  else
+  {
+    ecma_string_t *separator_string_p = ecma_get_string_from_value (separator_value);
+
+    /* 7-8. */
+    ecma_value_t first_value = ecma_op_typedarray_get_to_string_at_index (obj_p, 0);
+    if( ECMA_IS_VALUE_ERROR (first_value))
+    {
+      return first_value;
+    }
+
+    ecma_string_t *return_string_p = ecma_get_string_from_value (first_value);
+    ecma_ref_ecma_string (return_string_p);
+
+    /* 9-10. */
+    for (uint32_t k = 1; ecma_is_value_empty (ret_value) && (k < length); k++)
```

Comment:
This loop does not change `ret_value` thereby the function call `ecma_is_value_empty (ret_value)` in every iteration is unnecessary. This loop can be merged with the if statement below.